File: decompiled/4.0.2/be/ehealth/businessconnector/therlink/domain/responses/TherapeuticLinkResponseError.java
package be.ehealth.businessconnector.therlink.domain.responses;

public class TherapeuticLinkResponseError {
   private String errorCode;
   private String errorDescription;

   public TherapeuticLinkResponseError() {
   }

   public String getErrorCode() {
      return this.errorCode;
   }

   public void setErrorCode(String errorCode) {
      this.errorCode = errorCode;
   }

   public String getErrorDescription() {
      return this.errorDescription;
   }

   public void setErrorDescription(String errorDescription) {
      this.errorDescription = errorDescription;
   }

   public static class Builder {
      private TherapeuticLinkResponseError error = new TherapeuticLinkResponseError();

      public Builder() {
      }

      public Builder withErrorCode(String code) {
         this.error.setErrorCode(code);
         return this;
      }

      public Builder withErrorDescription(String desc) {
         this.error.setErrorDescription(desc);
         return this;
      }

      public TherapeuticLinkResponseError build() {
         return this.error;
      }
   }
}

File: src/main/kotlin/org/taktik/connector/business/therlink/domain/responses/TherapeuticLinkResponseError.kt
package org.taktik.connector.business.therlink.domain.responses

class TherapeuticLinkResponseError {
	var errorCode: String? = null
	var errorDescription: String? = null

	class Builder {
		private val error = TherapeuticLinkResponseError()

		fun withErrorCode(code: String): TherapeuticLinkResponseError.Builder {
			this.error.errorCode = code
			return this
		}

		fun withErrorDescription(desc: String): TherapeuticLinkResponseError.Builder {
			this.error.errorDescription = desc
			return this
		}

		fun build(): TherapeuticLinkResponseError {
			return this.error
		}
	}
}
